src/main/java/com/GG/T9AgeCombat/predicates/DetermineModificationPredicate.java

```diff
@@ -13,43 +13,86 @@ private DetermineModificationPredicate() {
     public static void applyBonus(OffensiveProfile offensiveProfile, ModificationEnum modification, Object value, Unit unitProfile, boolean isPermanent) {
         switch (modification) {
             case ATTACKS:
-                offensiveProfile.updateAttacks((int) value);
+                if(isPermanent) {
+                    offensiveProfile.updateAttacks((int) value);
+                } else {
+                    offensiveProfile.updateAttacksModifier((int) value);
+                }
```

Comment:
Minor but I'm wondering if it would look better if we instead move this to the profile.

For example we call an updateAttacks(value, permanent) and move the if/else logic to the update? Just wondering if this cleans up the code more by reducing the number of if/else around this.